### PR TITLE
[DEV-1140] manage unconfirmed user with expired code

### PR DESCRIPTION
**Pull request by MarcoPonchia**
#### List of Changes

- allow re-send of verification email for unconfirmed user on singup page
- manage error and allow re-send of verification email for unconfirmed user on confirmation page

#### Motivation and Context

manage unconfirmed user with expired code

#### How Has This Been Tested?

manually on:
- http://localhost:3000/auth/sign-up?email={email}&step=CONFIRM_SIGN_UP
- http://localhost:3000/auth/confirmation?username={username}&code={code}

#### Screenshots (if appropriate):
Screenshot from 2023-11-24 15-30-10

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/app/auth/confirmation/page.tsx [hunk 1]
```diff
@@ -1,56 +1,90 @@
...
       Auth.confirmSignUp(username, code)
         .then(() => {
-          setRenderNotFound(false);
           // eslint-disable-next-line functional/immutable-data
           router.push('/auth/account-activated');
         })
-        .catch(() => {
-          setRenderNotFound(true);
+        .catch((error) => {
+          // TODO: remove console warn and handle errors: [CodeMismatchException, ExpiredCodeException, InternalErrorException, LimitExceededException]
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/app/auth/confirmation/page.tsx:39]: this is TODO is relative to this issue https://pagopa.atlassian.net/browse/DEV-1179

File: apps/nextjs-website/src/components/organisms/Auth/AccountActivatedCard.tsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ const AccountActivatedCard = () => {
     auth: { accountActivated },
   } = translations;
 
+  // TODO: refactor using SingleCard component
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/organisms/Auth/AccountActivatedCard.tsx:12]: this is TODO is relative to this issue https://pagopa.atlassian.net/browse/DEV-1178

File: apps/nextjs-website/src/components/atoms/FileErrorIcon/FileErrorIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { SvgIcon } from '@mui/material';
+
+const FileErrorIcon = () => {
```
Comment (kin0992) [inline apps/nextjs-website/src/components/atoms/FileErrorIcon/FileErrorIcon.tsx:3]: Here I think we can safely use the `IllusError` icon coming from MUI-Italia.
https://github.com/pagopa/mui-italia/blob/main/src/illustrations/Error.tsx

File: apps/nextjs-website/src/app/auth/confirmation/page.tsx [hunk 4]
```diff
@@ -1,56 +1,90 @@
...
   const code = searchParams.get('code');
+  const t = useTranslations('auth');
 
-  const [renderNotFound, setRenderNotFound] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+  const [state, setState] = useState<State>(State.loading);
 
   useEffect(() => {
     if (username && code) {
+      console.log(username, code);
```
Comment (datalek) [inline apps/nextjs-website/src/app/auth/confirmation/page.tsx:32]:
```suggestion
```

File: apps/nextjs-website/src/app/auth/confirmation/page.tsx [hunk 5]
```diff
@@ -1,56 +1,90 @@
...
-        .catch(() => {
-          setRenderNotFound(true);
+        .catch((error) => {
+          // TODO: remove console warn and handle errors: [CodeMismatchException, ExpiredCodeException, InternalErrorException, LimitExceededException]
+          !isProduction && console.warn(error);
+          setState(State.resendCode);
         });
     } else {
-      setRenderNotFound(true);
+      console.log('username, code not found');
```
Comment (datalek) [inline apps/nextjs-website/src/app/auth/confirmation/page.tsx:44]:
```suggestion
```

**Review by datalek — CHANGES_REQUESTED** [review]

**Review by datalek — APPROVED** [review]
I've opened two issues [DEV-1183](https://pagopa.atlassian.net/browse/DEV-1183) and [DEV-1184](https://pagopa.atlassian.net/browse/DEV-1184)

**Review by kin0992 — APPROVED** [review]

**Pull request merged**